website/content/docs/intro/use-cases/blue-green-deployment.mdx

```diff
@@ -0,0 +1,447 @@
+the existing version in an environment. While both versions are running, DNS is
+updated to split traffic flowing to the application between the new and existing
+versions of the app. This enables essentially nil downtime for the application
+during deployments in most cases, but also enables operators to provide the experience
+of the new version of an application to a certain percentage of users, without
+forcing all users to use the newer version. Depending upon feedback and testing
+while the new version is live and processing a partial amount of traffic, a decision
+is made to either promote or roll back the new version, and DNS is again updated
+to route 100% of traffic to all instances running the application.
+
+In the HashiStack, this is made possible with features from the workload orchestration
+tool, Hashicorp Nomad, and the service networking tool, Hashicorp Consul. Nomad
+enables operators to deploy [canary](https://developer.hashicorp.com/nomad/docs/job-specification/update#canary)
+instances of a newer version of an application, without destroying the existing
+instances. The canary deployment can be auto-promoted when "healthy", or manually
+promoted by operators. Consul plays a role as well, through service configuration
+entries. The [service resolver](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-resolver)
+config entry controls which instances of a service should be matched by downstream
+requests made to the service, which can be [filtered](https://developer.hashicorp.com/consul/api-docs/features/filtering)
+by service tags. A [service splitter](https://developer.hashicorp.com/consul/docs/connect/config-entries/service-splitter)
+config entry splits a configured percentage of traffic to resolvers. These features
+of Nomad and Consul work together by the canary instances of a new Nomad job version
+registering new instances of a Consul service with [canary tags](https://developer.hashicorp.com/nomad/docs/job-specification/service#canary_tags).
+The tags applied to the canary instances of the service match a service resolver.
+The service splitter is then updated while both the new and old application instances
+are running to split some percentage of traffic between the resolver for the new
+and old instances of the service.
+
+To achieve this, a series of commands or API calls would need to be run manually,
+or automated through scripts or some other means; but Waypoint pipelines can orchestrate
+this! Waypoint pipelines can execute configured phases of your app's build, deployment,
+and release lifecycle, but also run custom steps, such as modifying Consul config
+entries to configure DNS to route traffic to your canary deployment instances.
+
+An example where this is configured is located [here](https://github.com/paladin-devops/hashitalks-deploy-blue-green).
+The Waypoint configuration file contains three pipelines, and each one will be
+described in the next section.
+
```

Comment:
Before we dive deep into the example, it might be nice to have a small paragraph or bulleted list of required services/tools that need to be configured to follow along. i.e. A Nomad and Consul cluster that Waypoint is connected to, etc. Just so the reader knows what kind of things they might need. Not like a full setup guide, but maybe like a heads up you might want to be familiar with these tools before going forward sort of thing?

Reply:
Added [here](https://github.com/hashicorp/waypoint/pull/4263/commits/cb9a63bd782a634d22fbc0480c0b03896d1eee34)!